example/lib/main.dart

```diff
@@ -16,25 +13,22 @@ void main() {
 final RouteObserver<PageRoute> routeObserver = RouteObserver<PageRoute>();
 
 class MyApp extends StatelessWidget {
+  final GlobalKey<NavigatorState> _navigatorKey = GlobalKey();
+
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'FlutterBleLib example',
-      theme: new ThemeData(
-        primaryColor: new Color(0xFF0A3D91),
-        accentColor: new Color(0xFFCC0000),
+    return BlocProvider<NavigatorBloc>(
+      builder: (context) => NavigatorBloc(navigatorKey: _navigatorKey),
+      child: MaterialApp(
+        navigatorKey: _navigatorKey,
+        title: 'FlutterBleLib example',
+        theme: new ThemeData(
+          primaryColor: new Color(0xFF0A3D91),
```

Comment:
Perhaps those will only be used here, but it might be wise to extract them to separate file as named constants.

Reply:
Extracted to a separate file. Made some minor style adjustments (that also belong to that separated file) along the way.